### PR TITLE
adding QFX5100-24Q-2P

### PR DESCRIPTION
**Pull request by sjtarik**
adding a new Juniper Device  QFX5100-24Q-2P

### CODE REVIEW

**Comment by danner26** [comment]
Please append `juniper` to the slug
Is the `q` at the end of model suppose to be there?
Please add `weight` and `airflow` if you have it
Please add a comment with the products datasheet (if availible) in markdown format
It looks like this device has 2 module bays in the front of it, is that true? If so, please add them
It looks like the fans and the PSUs are modular. Is this true? If so, please add them in module bays as well. 

Thanks!

**Comment by sjtarik** [comment]
@danner26 thank you for the detailed review. I think I got most of them but can you take a look again?

**Comment by danner26** [comment]
Hey @sjtarik this is perfect! Thank you
The last thing is that since the PSUs are modular, you do not need the `power-ports` anymore. Both of these can be removed. Instead, the PSUs which are to be installed here should be modelled within the module-types folder if they are not there already. Please remove the `power-ports` and then we can merge this, or if you want to add PSU module-types I can review those as well!

File: device-types/Juniper/QFX5100-24Q-2P.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    type: iec-60320-c14
+    maximum_draw: 650
+module-bays:
+  - name: Fan Tray 0
+  - name: Fan Tray 1
+  - name: Fan Tray 2
+  - name: Fan Tray 3
+  - name: Fan Tray 4
+  - name: PIC1
+    label: Expansion module QFX-EM-4Q
```
Comment (jsenecal) [inline device-types/Juniper/QFX5100-24Q-2P.yaml:86]: These are not the only modules that can fit in there. Please be more generic in your labels.
image

File: device-types/Juniper/QFX5100-24Q-2P.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    position: PIC1
+  - name: PIC2
+    label: Expansion module QFX-EM-4Q
```
Comment (jsenecal) [inline device-types/Juniper/QFX5100-24Q-2P.yaml:89]: Same as above

File: device-types/Juniper/QFX5100-24Q-2P.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+---
+manufacturer: Juniper
+model: QFX5100-24Q-2P
```
Comment (jsenecal) [inline device-types/Juniper/QFX5100-24Q-2P.yaml:3]: The `-2P` variant is not mentioned anywhere in juniper's documentation - where is this from ?
https://www.juniper.net/content/dam/www/assets/datasheets/us/en/switches/qfx5100-ethernet-switch-datasheet.pdf

File: device-types/Juniper/QFX5100-24Q-2P.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/23
+    type: 40gbase-x-qsfpp
+  - name: em0
+    type: 1000base-t
+    mgmt_only: true
+  - name: em1
+    type: 1000base-x-sfp
+    mgmt_only: true
+power-ports:
```
Comment (jsenecal) [inline device-types/Juniper/QFX5100-24Q-2P.yaml:72]: Please remove (see https://github.com/netbox-community/devicetype-library/pull/1140#issuecomment-1441828737)

**Review by jsenecal — CHANGES_REQUESTED** [review]
Please also submit appropriate modules for this device if possible. Thanks!

File: device-types/Juniper/QFX5100-24Q-2P.yaml (hunk 3 above)
Comment (sjtarik) [reply device-types/Juniper/QFX5100-24Q-2P.yaml:3]: Not in the product sheet but from device itself. Which one you go with usually?

```
> show chassis hardware
Hardware inventory:
Item             Version  Part number  Serial number     Description
Chassis                                XXXXXXXXXXXX      QFX5100-24Q-2P
Pseudo CB 0
Routing Engine 0          BUILTIN      BUILTIN           QFX Routing Engine
```

**Comment by sjtarik** [comment]
Screenshot 2023-02-24 at 4 11 47 PM

Not sure how this happened, I just clicked on re-request review from @jsenecal and it removed @jsenecal's review request from @danner26 Lol

**Comment by danner26** [comment]
> Screenshot 2023-02-24 at 4 11 47 PM
> 
> Not sure how this happened, I just clicked on re-request review from @jsenecal and it removed @jsenecal's review request from @danner26 Lol

No worries. I am keeping an eye on this as well

**Review by jsenecal — CHANGES_REQUESTED** [review]
You havent added the modules in that PR. Please provide them as well.

File: device-types/Juniper/QFX5100-24Q-2P.yaml (hunk 3 above)
Comment (jsenecal) [reply device-types/Juniper/QFX5100-24Q-2P.yaml:3]: try with `show chassis hardware models` and `show chassis hardware clei-models` :)
In any case I think we should go with the QFX5100-24Q but I'm waiting for this first.

Comment (sjtarik) [reply device-types/Juniper/QFX5100-24Q-2P.yaml:3]:
```
> show chassis hardware models
error: command is not valid on the qfx5100-24q-2p
```

```
> show chassis hardware clei-models
Hardware inventory:
Item             Version  Part number  CLEI code         FRU model number
Routing Engine 0          BUILTIN      XXX        QFX5100-24Q-AFO
FPC 0            REV 28   650-049942   XXX        QFX5100-24Q-AFO
  PIC 0                   BUILTIN      XXX        QFX5100-24Q-AFO
  PIC 1          REV 02   611-064275   XXX        EX4600-EM-8F
  PIC 2          REV 02   611-064275   XXX        EX4600-EM-8F
Power Supply 0   REV 03   740-041741   XXX        JPSU-650W-AC-AFO
Power Supply 1   REV 03   740-041741   XXX        JPSU-650W-AC-AFO
Fan Tray 0                                               QFX5100-FANAFO
Fan Tray 1                                               QFX5100-FANAFO
Fan Tray 2                                               QFX5100-FANAFO
Fan Tray 3                                               QFX5100-FANAFO
Fan Tray 4                                               QFX5100-FANAFO
```

Comment (sjtarik) [reply device-types/Juniper/QFX5100-24Q-2P.yaml:3]: So there is one other model "QFX5100-48S-3AFO.yaml" that has "AFO" in model name, should I drop AFO?

**Comment by danner26** [comment]
I am going to let @jsenecal head this one up since he has a lot more expertise with these than me (I haven't used juniper devices in any of my jobs).

**Comment by danner26** [comment]
Hello, unfortunately today we had to make a change to the schema in which weight is defined by. The upstream NetBox API requires weight to be passed to it in the following format:
```
weight: 12.01
weight_unit: lb
```

Due to this change, I have updated your branch against the current master. This might require you to update your PR, if you had weight currently defined, so that your PR is in-line with the new weight schema. If you have any issues, please let me know or refer to [this PR](https://github.com/netbox-community/devicetype-library/pull/1187) for more information.

**Comment by sjtarik** [comment]
@danner26 @jsenecal thanks for the memo. weight unit issue fixed now. can you folks review and merge this if it is ok? modules/bay devices not really needed in my use case since I am not able to list the interfaces of children modules or assign IP addresses to them.

File: device-types/Juniper/QFX5100-24Q-2P.yaml (hunk 3 above)
Comment (sjtarik) [reply device-types/Juniper/QFX5100-24Q-2P.yaml]: there is only on device that is named after the direction of air/inlet. I guess it makes much more sense to name the device without it.

**Comment by jsenecal** [comment]
Hi @sjtarik, unfortunately you still haven't added the modules to this PR, such as power supplies, or SFPP modules.
You can use these as examples:
https://github.com/netbox-community/devicetype-library/blob/master/module-types/Juniper/SRX600-PWR-645AC-%20POE.yaml
https://github.com/netbox-community/devicetype-library/blob/master/module-types/Juniper/EX-UM-4x4SFP.yaml

**Comment by sjtarik** [comment]
Sorry, priorities changed for my project. Closing the pull request to remove clutter but will reopen if I start actively working on it. Thanks,TarikOn Mar 13, 2023, at 6:44 AM, Jonathan Senecal ***@***.***> wrote:﻿
Hi @sjtarik, unfortunately you still haven't added the modules to this PR, such as power supplies, or SFPP modules.
You can use these as examples:
https://github.com/netbox-community/devicetype-library/blob/master/module-types/Juniper/SRX600-PWR-645AC-%20POE.yaml
https://github.com/netbox-community/devicetype-library/blob/master/module-types/Juniper/EX-UM-4x4SFP.yaml

**Pull request closed without merging**